firefox-ios/Client/Frontend/TrackingProtection/BlockedTrackersTableController.swift

```diff
@@ -0,0 +1,385 @@
+
+        return [
+            BlockedTrackerItem(
+                title: crossSiteText,
+                image: crossSiteImage
+            ),
+            BlockedTrackerItem(
+                title: fingerprintersText,
+                image: fingerprintersImage
+            ),
+            BlockedTrackerItem(
+                title: trackingContentText,
+                image: trackingContentImage
+            ),
+            BlockedTrackerItem(
+                title: socialMediaText,
+                image: socialMediaImage
+            )
+        ]
+    }
+}
+
+// MARK: BlockedTrackersTableViewController
+class BlockedTrackersTableViewController: UIViewController,
+                                          Themeable,
+                                          UITableViewDelegate {
+    private lazy var blockedTrackersTableView: UITableView = {
+        let blockedTrackersTable = UITableView(frame: .zero, style: .insetGrouped)
+        blockedTrackersTable.translatesAutoresizingMaskIntoConstraints = false
+        blockedTrackersTable.showsHorizontalScrollIndicator = false
+        blockedTrackersTable.backgroundColor = .clear
+        blockedTrackersTable.delegate = self
+        blockedTrackersTable.layer.cornerRadius = TPMenuUX.UX.viewCornerRadius
+        blockedTrackersTable.allowsSelection = false
+        blockedTrackersTable.separatorColor = .clear
+        blockedTrackersTable.separatorStyle = .singleLine
+        blockedTrackersTable.isScrollEnabled = false
+        blockedTrackersTable.showsVerticalScrollIndicator = false
+        blockedTrackersTable.rowHeight = UITableView.automaticDimension
+        blockedTrackersTable.estimatedRowHeight = TPMenuUX.UX.BlockedTrackers.estimatedRowHeight
+        blockedTrackersTable.estimatedSectionHeaderHeight = TPMenuUX.UX.BlockedTrackers.headerPreferredHeight
+        blockedTrackersTable.register(
+            BlockedTrackerCell.self,
+            forCellReuseIdentifier: BlockedTrackerCell.cellIdentifier
+        )
+        blockedTrackersTable.register(
+            BlockedTrackersHeaderView.self,
+            forHeaderFooterViewReuseIdentifier: BlockedTrackersHeaderView.cellIdentifier
+        )
+        return blockedTrackersTable
+    }()
+    private var dataSource: UITableViewDiffableDataSource<Int, BlockedTrackerItem>!
+    private var headerView: BlockedTrackersHeaderView!
+
+    // MARK: Navigation View
+    // TODO: Replace this with the HeaderView defined for the TrackingDetailsViewController
+    private let navigationView = UIView()
+    private let horizontalLine: UIView = .build { _ in }
+    private let siteTitleLabel: UILabel = .build { label in
+        label.adjustsFontForContentSizeCategory = true
+        label.textAlignment = .center
+        label.font = FXFontStyles.Regular.headline.scaledFont()
+        label.numberOfLines = 2
+        label.accessibilityTraits.insert(.header)
+    }
+    private var closeButton: UIButton = .build { button in
```

Comment:
why not use the `CloseButton` from the component library

Reply:
it will use the custom view from the tracking protection details view instead of this